### ISSUE TITLE
Support SRV record resolving

### ISSUE THREAD

**Issue by zedd3v** [sindresorhus/got#1478]
#### What problem are you trying to solve?

Supporting SRV DNS records

#### Describe the feature

Add SRV option for requests which will resolve the DNS and use it or fallback to the original domain if none is found.

Example: https://github.com/JonGretar/srv-request/blob/master/index.js

#### Checklist

- [✔️ ] I have read the documentation and made sure this feature doesn't already exist.

**Comment by szmarczak** [comment]
I doubt that this will be supported in Got core, I'm not really sure. This is better suited for Node.js or [`cacheable-lookup`](https://github.com/szmarczak/cacheable-lookup/blob/master/source/index.js) imo.

/cc @sindresorhus

**Comment by Giotino** [comment]
I don't think querying for a SRV record on the hostname makes any sense.

SRV record is defined by https://tools.ietf.org/html/rfc2782

Its format is _Service._Proto.Name (eg. _sip._udp.example.com), and it contains the address and the port of the related service. When you make an HTTP request you already need to know the service address and port.
If you define a service with a SRV record (eg. _MyHttpService._tcp.example.com) Got has no way to know how the service you want to use is called when you make a request like `got("http://example.com")`, I don't think Got should implement something to have the user specify a service name to search for.

TL; DR: You should fetch the service address and port yourself before marking the request. Got cannot read your mind.

This answer was a little generic, after this you can find you case specific answer.

Reading your example you seem to want to make a request straight to `_MyHttpService._tcp.example.com` but, even if Got would support something like that, it would need an option to specify to use the SRV record. This cannot be a default behavior.

All that said I think Got should handle only A and AAAA records.

**Comment by szmarczak** [comment]
Yep. There may be many HTTP services out there.

**Comment by szmarczak** [comment]
@Giotino The `lookup` option is not documented. Would you be interested in sending a PR?

**Comment by szmarczak** [comment]
I'm closing. SRV is too general to be used by Got. The service name doesn't have to be `http`, `https`, `h2`. It can also be `myHttpServer` etc...
